tests/test_configuration.py

```diff
@@ -362,3 +365,138 @@ def test_validate_unknown_config_option(self):
+            '[bugsnag] CRITICAL - hello critical!\n'
+        )
+
+        config.logger.exception('hello exception! %s', Exception('oh no!'))
+        assert mock_stderr.getvalue().endswith(
+            '[bugsnag] ERROR - hello exception! oh no!\nNoneType: None\n'
+        )
+
+    @patch('sys.stderr', new_callable=StringIO)
+    def test_default_logger_with_configure_call(self, mock_stderr):
+        config = Configuration()
+        config.configure(app_type='very good')
+
+        assert isinstance(config.logger, logging.Logger)
+
+        # debug and info calls should be ignored by default
+        config.logger.debug('hello debug!')
+        config.logger.info('hello info!')
+        assert mock_stderr.getvalue() == ''
+
+        config.logger.warning('hello warning!')
+        assert mock_stderr.getvalue().endswith(
+            '[bugsnag] WARNING - hello warning!\n'
+        )
+
+        config.logger.error('hello error!')
+        assert mock_stderr.getvalue().endswith(
+            '[bugsnag] ERROR - hello error!\n'
+        )
+
+        config.logger.critical('hello critical!')
+        assert mock_stderr.getvalue().endswith(
+            '[bugsnag] CRITICAL - hello critical!\n'
+        )
+
+        config.logger.exception('hello exception! %s', Exception('oh no!'))
+        assert mock_stderr.getvalue().endswith(
+            '[bugsnag] ERROR - hello exception! oh no!\nNoneType: None\n'
+        )
+
+    @patch('sys.stderr', new_callable=StringIO)
+    def test_logger_can_be_disabled(self, mock_stderr):
+        config = Configuration(logger=None)
+
+        assert isinstance(config.logger, logging.Logger)
+
+        config.logger.debug('hello debug!')
+        config.logger.info('hello info!')
+        config.logger.warning('hello warning!')
+        config.logger.error('hello error!')
+        config.logger.critical('hello critical!')
+        config.logger.exception('hello exception!')
+
+        assert mock_stderr.getvalue() == ''
+
+    @patch('sys.stderr', new_callable=StringIO)
+    def test_logger_can_be_disabled_using_configure(self, mock_stderr):
+        config = Configuration()
+        config.configure(logger=None)
+
+        assert isinstance(config.logger, logging.Logger)
+
+        config.logger.debug('hello debug!')
+        config.logger.info('hello info!')
+        config.logger.warning('hello warning!')
+        config.logger.error('hello error!')
+        config.logger.critical('hello critical!')
+        config.logger.exception('hello exception!')
+
+        assert mock_stderr.getvalue() == ''
+
+    @patch('sys.stderr', new_callable=StringIO)
+    def test_default_logger_is_used_if_invalid_logger_set(self, mock_stderr):
+        with pytest.warns(RuntimeWarning) as warnings:
+            config = Configuration(logger=123)
+
+        expected_message = 'logger should be logging.Logger, got int'
+
+        assert len(warnings) == 1
+        assert warnings[0].message.args[0] == expected_message
```

Comment:
Shouldn't this be inside the context at line 465 to access `warnings`? Seems out that its accessible outside of that scope.

Reply:
`with` doesn't create a scope so it's accessible here; this is straight out of the docs: https://docs.pytest.org/en/stable/warnings.html#warns

Reply:
ah! 💡